### PR TITLE
add suggest top10 easy meals use case test

### PR DESCRIPTION
**Pull request by raouf-kamel**
## 📌 What was added?
A new test class `SuggestTop10EasyMealsUseCaseTest` was created to validate the behavior of the use case.

---

## ✅ Covered Scenarios:
- [x] Should return all meals when the filtered list is less than 10.
- [x] Should return only 10 meals when the filtered list is greater than 10.
- [x] Should return exactly 10 meals when the filtered list matches the limit.
- [x] Should skip meals with `null` values (e.g., `preparationTime`, `numberOfIngredients`, `numberOfSteps`).
- [x] Should throw `EmptyDataException` when the meals list is empty.

---

## 🤔 Why is this important?
This ensures that the `SuggestTop10EasyMealsUseCase` handles edge cases gracefully and returns reliable results, even with partial or invalid data.

---

## 🚀 How to test?
Run the unit tests in `SuggestTop10EasyMealsUseCaseTest` class:
```bash
./gradlew test

### CODE REVIEW

File: src/test/kotlin/logic/usecase/SuggestTop10EasyMealsUseCaseTest.kt [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        every { mealsRepository.getAllMeals() } returns meals
+
+        val result = getSuggestTop10EasyMealsUseCase.suggestEasyMeals()
+
+
+        assertThat(result.size).isAtMost(10)
+
+        result.forEach { meal ->
+            assertThat(meal.preparationTime).isNotNull()
+            assertThat(meal.preparationTime!!).isAtMost(30)
```
Comment (mohamedd-hassan) [inline src/test/kotlin/logic/usecase/SuggestTop10EasyMealsUseCaseTest.kt]: i notice here that you are creating a meal with null preparation time while at the same time asserting that its not null

**Review by mohamedd-hassan — CHANGES_REQUESTED** [review]
its not really clear if this is being done already but try adding a test case where some of the meals do not fit the (the preparation time is too long or the number of ingredients is more than 6) and see if it will get filtered or not

File: src/test/kotlin/logic/usecase/SuggestTop10EasyMealsUseCaseTest.kt [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+import logic.exception.FoodMoodException
+import logic.helpers.createMeal
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+
+class SuggestTop10EasyMealsUseCaseTest {
+    private lateinit var mealsRepository: MealsRepository
+    private lateinit var getSuggestTop10EasyMealsUseCase: SuggestTop10EasyMealsUseCase
```
Comment (Belal-Alkhatib) [inline src/test/kotlin/logic/usecase/SuggestTop10EasyMealsUseCaseTest.kt]: @MohammadAshrf suggested simplifying the variable names to reduce redundancy and improve readability in the test cases

So I suggest changing the naming here to be just:

```
private lateinit var repository: MealsRepository
private lateinit var useCase: SuggestTop10EasyMealsUseCase
```
To align with our naming conventions and maintain consistency across the codebase

File: src/test/kotlin/logic/usecase/SuggestTop10EasyMealsUseCaseTest.kt [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    @Test
+    fun `should returns all meals when filtered list size is less than 10`() {
+        //Given
+        val meals = (1..5).map {
+            createMeal(
+                name = "Meal $it",
+                preparationTime = 20,
+                numberOfIngredients = 4,
+                numberOfSteps = 5
+            )
```
Comment (Belal-Alkhatib) [inline src/test/kotlin/logic/usecase/SuggestTop10EasyMealsUseCaseTest.kt]: To make the class even cleaner, consider moving the constants and dummy data to a separate SuggestTop10EasyMealsMealTestFactory object for example. This will keep the test cases focused.

File: src/test/kotlin/logic/usecase/SuggestTop10EasyMealsUseCaseTest.kt [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    fun `should returns all meals when filtered list size is exactly 10`() {
+        //Given
+        val meals = (1..10).map {
+            createMeal(
+                name = "Meal $it",
+                preparationTime = 20,
+                numberOfIngredients = 4,
+                numberOfSteps = 5
+            )
+        }
```
Comment (Belal-Alkhatib) [inline src/test/kotlin/logic/usecase/SuggestTop10EasyMealsUseCaseTest.kt]: The same `createMeal(...)` logic appears in multiple tests. You could extract helper function you need to your TestFactory object

File: src/test/kotlin/logic/usecase/SuggestTop10EasyMealsUseCaseTest.kt [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+        every { mealsRepository.getAllMeals() } returns meals
+
+        //when
+        val result = getSuggestTop10EasyMealsUseCase.suggestEasyMeals()
+        //then
+        assertEquals(5, result.size)
```
Comment (Belal-Alkhatib) [inline src/test/kotlin/logic/usecase/SuggestTop10EasyMealsUseCaseTest.kt:38]: You're using both:

```
org.junit.jupiter.api.Assertions.assertEquals
com.google.common.truth.Truth.assertThat
```

Pick one assertion style across the class for consistency " I prefer `Truth.assertThat()` for more readable and expressive assertions ".

File: src/test/kotlin/logic/usecase/SuggestTop10EasyMealsUseCaseTest.kt [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+
+        val result = getSuggestTop10EasyMealsUseCase.suggestEasyMeals()
+
+
+        assertThat(result.size).isAtMost(10)
+
+        result.forEach { meal ->
+            assertThat(meal.preparationTime).isAtMost(30)
+            assertThat(meal.numberOfIngredients).isAtMost(5)
+            assertThat(meal.numberOfSteps).isAtMost(6)
```
Comment (Belal-Alkhatib) [inline src/test/kotlin/logic/usecase/SuggestTop10EasyMealsUseCaseTest.kt]: - split large test case into multiple smaller, focused tests, each checking a specific condition (e.g., meal size, preparation time, ingredients, and steps).

- replaced `forEach `iteration with `containsAtLeast `assertions for more concise and expressive validation.

- ensured each test has a single responsibility, improving clarity and reducing complexity.

File: src/main/kotlin/logic/usecase/SuggestTop10EasyMealsUseCase.kt
Comment (Belal-Alkhatib) [inline src/main/kotlin/logic/usecase/SuggestTop10EasyMealsUseCase.kt:1]: could you please remove the extra line spaces? It’ll help keep the code neat and easier to read. 🙌

**Review by Belal-Alkhatib — CHANGES_REQUESTED** [review]

**Review by mohamedd-hassan — APPROVED** [review]
lgtm

**Review by Belal-Alkhatib — CHANGES_REQUESTED** [review]
Please ensure all convention-related issues are resolved before submitting the review request

File: src/main/kotlin/logic/usecase/SuggestTop10EasyMealsUseCase.kt
Comment (raouf-kamel) [reply src/main/kotlin/logic/usecase/SuggestTop10EasyMealsUseCase.kt:1]: Really! Without you, we would be lost in the blank lines. Thank you, Bilal♥ .

**Review by mohamedd-hassan — APPROVED** [review]

**Review by MohammadAshrf — APPROVED** [review]
nice
